openpype/modules/shotgrid/plugins/publish/collect_shotgrid_entities.py

```diff
@@ -0,0 +1,102 @@
+import os
+
+import pyblish.api
+from openpype.lib.mongo import OpenPypeMongoConnection
+
+from openpype.modules.shotgrid.lib.settings import (
+    get_shotgrid_project_settings,
+)
+
+
+class CollectShotgridEntities(pyblish.api.ContextPlugin):
+    """Collect shotgrid entities according to the current context"""
+
+    order = pyblish.api.CollectorOrder + 0.499
+    label = "Shotgrid entities"
+
+    def process(self, context):
+
+        avalon_project = context.data.get("projectEntity")
+        avalon_asset = context.data.get("assetEntity")
+        avalon_task_name = os.getenv("AVALON_TASK")
+
+        self.log.info(avalon_project)
+        self.log.info(avalon_asset)
+
+        sg_project = _get_shotgrid_project(avalon_project)
+        sg_task = _get_shotgrid_task(
+            avalon_project, avalon_asset, avalon_task_name
+        )
+        sg_entity = _get_shotgrid_entity(avalon_project, avalon_asset)
+
+        if sg_project:
+            context.data["shotgridProject"] = sg_project
+            self.log.info(
+                "Collected correspondig shotgrid project : {}".format(
+                    sg_project
+                )
+            )
+
+        if sg_task:
+            context.data["shotgridTask"] = sg_task
+            self.log.info(
+                "Collected correspondig shotgrid task : {}".format(sg_task)
+            )
+
+        if sg_entity:
+            context.data["shotgridEntity"] = sg_entity
+            self.log.info(
+                "Collected correspondig shotgrid entity : {}".format(sg_entity)
+            )
+
+    def _find_existing_version(self, code, context):
+
+        filters = [
+            ["project", "is", context.data.get("shotgridProject")],
+            ["sg_task", "is", context.data.get("shotgridTask")],
+            ["entity", "is", context.data.get("shotgridEntity")],
+            ["code", "is", code],
+        ]
+
+        sg = context.data.get("shotgridSession")
+        return sg.find_one("Version", filters, [])
+
+
+def _get_shotgrid_collection(project):
+    client = OpenPypeMongoConnection.get_mongo_client()
+    return client.get_database("shotgrid_openpype").get_collection(project)
+
+
+def _get_shotgrid_project(avalon_project):
+    proj_settings = get_shotgrid_project_settings(avalon_project["name"])
+    shotgrid_project_id = proj_settings.get("shotgrid_project_id")
+    if shotgrid_project_id:
+        return {"type": "Project", "id": shotgrid_project_id}
+    return {}
+
+
+def _get_shotgrid_task(avalon_project, avalon_asset, avalon_task):
```

Comment:
I have few questions regarding this function.
1. Why is shotgrid creating separated mongo database for data that should be in avalon database?
    At least it looks it's done that way. I maybe missed that part of shotgrid implementation?
2. Why `_id` contains strings?

Few comments:
Please try avoid using `avalon` in variables and also be more specific about the content (e.g. `asset_doc` or `task_name` tells more about it's content), also pass project document just to use it's name can be simplified to pass just the name?

Reply:
In my memory, it seemed simpler ans more maintainable for @victorbartel to store shotgrid data in another database and store in avalon on the asset that corresponds to the id of this database